java/core/src/main/java/sleeper/core/statestore/SplitFileReferenceRequest.java

```diff
@@ -27,12 +27,13 @@ public class SplitFileReferenceRequest {
     private final FileReference oldReference;
     private final List<FileReference> newReferences;
 
-    public SplitFileReferenceRequest(FileReference oldReference, List<FileReference> newReferences) {
-        this.oldReference = Objects.requireNonNull(oldReference, "oldReference must not be null");
-        this.newReferences = Objects.requireNonNull(newReferences, "newReferences must not be null");
+    public static SplitFileReferenceRequest from(FileReference oldReference, List<FileReference> newReferences) {
         if (newReferences.isEmpty()) {
             throw new IllegalArgumentException("newReferences must not be empty");
```

Comment:
This has been moved to before the null check. The null check needs to happen first.